src/tests/secp256k1_blake160_sighash_all.rs

```diff
@@ -152,22 +179,89 @@ fn test_sighash_all_unlock() {
     let privkey = Generator::random_privkey();
     let pubkey = privkey.pubkey().expect("pubkey");
     let pubkey_hash = blake160(&pubkey.serialize());
-    let tx = gen_tx(&mut data_loader, SIGHASH_ALL_BIN.clone(), pubkey_hash);
+    let tx = gen_tx(&mut data_loader, pubkey_hash);
     let tx = sign_tx(tx, &privkey);
     let resolved_tx = build_resolved_tx(&data_loader, &tx);
     let verify_result =
         TransactionScriptsVerifier::new(&resolved_tx, &data_loader).verify(MAX_CYCLES);
     verify_result.expect("pass verification");
 }
 
+#[test]
+fn test_sighash_all_with_extra_witness_unlock() {
+    let mut data_loader = DummyDataLoader::new();
+    let privkey = Generator::random_privkey();
+    let pubkey = privkey.pubkey().expect("pubkey");
+    let pubkey_hash = blake160(&pubkey.serialize());
+    let tx = gen_tx(&mut data_loader, pubkey_hash);
+    let extract_witness = vec![1, 2, 3, 4];
+    let tx = tx
+        .as_advanced_builder()
+        .set_witnesses(vec![Bytes::from(extract_witness).pack()])
+        .build();
+    {
+        let tx = sign_tx(tx.clone(), &privkey);
+        let resolved_tx = build_resolved_tx(&data_loader, &tx);
+        let verify_result =
+            TransactionScriptsVerifier::new(&resolved_tx, &data_loader).verify(MAX_CYCLES);
+        verify_result.expect("pass verification");
+    }
+    {
+        let tx = sign_tx(tx, &privkey);
+        let wrong_witness = tx.witnesses().get(0).unwrap().as_builder().push(0).build();
+        let tx = tx
+            .as_advanced_builder()
+            .set_witnesses(vec![wrong_witness])
+            .build();
+        let resolved_tx = build_resolved_tx(&data_loader, &tx);
+        let verify_result =
+            TransactionScriptsVerifier::new(&resolved_tx, &data_loader).verify(MAX_CYCLES);
+        assert_error_eq!(
+            verify_result.unwrap_err(),
+            ScriptError::ValidationFailure(ERROR_PUBKEY_BLAKE160_HASH),
+        );
+    }
+}
+
+#[test]
+fn test_sighash_all_with_multiple_inputs_unlock() {
```

Comment:
This name is not proper IMO, can't we unlock multiple inputs tx success?

Can we add a test case for 2in-2out success unlock?